src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameInfo.cs

```diff
@@ -29,17 +29,14 @@ internal readonly struct NameDeclarationInfo(
 
     private static readonly ImmutableArray<SymbolKindOrTypeKind> s_propertySyntaxKind =
         [new SymbolKindOrTypeKind(SymbolKind.Property)];
-
-    private readonly ImmutableArray<SymbolKindOrTypeKind> _possibleSymbolKinds = possibleSymbolKinds;
-
     public readonly DeclarationModifiers Modifiers = declarationModifiers;
     public readonly Accessibility? DeclaredAccessibility = accessibility;
 
     public readonly ITypeSymbol? Type = type;
     public readonly IAliasSymbol? Alias = alias;
     public readonly ISymbol? Symbol = symbol;
 
-    public ImmutableArray<SymbolKindOrTypeKind> PossibleSymbolKinds => _possibleSymbolKinds.NullToEmpty();
+    public ImmutableArray<SymbolKindOrTypeKind> PossibleSymbolKinds { get => field.NullToEmpty(); } = possibleSymbolKinds;
```

Comment:
I know this PR didn't introduce it, but is there a quirk here that makes `NullToEmpty()` necessary?

Reply:
This is primarily making sure that if someone either passed `default` to the consturctor *or* just has a `default` instance of NameDeclarationInfo that PossibleSymbolKinds still always returns an empty array.  It makes it so that teh entire domain you have to think about for PossibleSymbolKinds is non-defualt arrays (so real arrays of 0 to N length).  It removes the null/default case as something to ever even have to eb concerned with.

Reply:
Practically, we have good hygiene, and this is generally not a concern.  But it is something that occasionally arises, and thse patterns help.